### PR TITLE
Add method to compute `GitShortHash` of a `GitObject`.

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]
Awesome!

**Comment by kshyatt** [comment]
Since we're in FF, can we merge this?

**Comment by ararslan** [comment]
Sounds like a question for @tkelman, supreme freezer of features

**Comment by tkelman** [comment]
what's this going to be used for?

**Comment by ararslan** [comment]
~~EVIL!~~ Getting the short hash for an object. IIRC staticfloat was looking for this exact functionality the other day.

**Comment by tkelman** [comment]
Right but nothing in base needs this, and is this something that absolutely needs to be exposed right now?

Is the length of what this outputs a specific size, or as short as possible while unambiguous for a given repo?

**Comment by ararslan** [comment]
> nothing in base needs this

Correct AFAIK

> is this something that absolutely needs to be exposed right now?

IMO no

> Is the length of what this outputs a specific size, or as short as possible while unambiguous for a given repo?

Hm, unclear in the upstream LibGit2 docs.

**Comment by StefanKarpinski** [comment]
Given that this is an unexported function, I can't see the harm in merging this.

**Comment by simonbyrne** [comment]
I've updated the docs to clarify what it does.

File: base/libgit2/oid.jl [hunk 1]
```diff
@@ -44,6 +67,15 @@ macro githash_str(id)
         GitHash(id)
     end
 end
+
+
+"""
+    GitHash(ref::GitReference)
+
+Gets the identifier (`GitHash`) of th object referred to by the direct reference
```
Comment (ararslan) [inline base/libgit2/oid.jl]: s/Gets/Get/ to stick to the imperative, s/th/the/

File: base/libgit2/oid.jl [hunk 2]
```diff
@@ -44,6 +67,15 @@ macro githash_str(id)
...
+`ref`. Note: this does not work for symbolic references, use `GitHash(repo::GitRepo,
+ref_name::AbstractString)` instead.
```
Comment (ararslan) [inline base/libgit2/oid.jl]: "instead for that case" is somewhat more explicit imo

File: base/libgit2/oid.jl [hunk 3]
```diff
@@ -52,6 +84,13 @@ function GitHash(ref::GitReference)
     return GitHash(oid_ptr)
 end
 
+
+"""
+    GitHash(repo::GitRepo, ref_name::AbstractString)
+
+Gets the identifier (`GitHash`) of the object referred to by reference specified by
```
Comment (ararslan) [inline base/libgit2/oid.jl]: Get

File: base/libgit2/oid.jl [hunk 4]
```diff
@@ -61,10 +100,31 @@ function GitHash(repo::GitRepo, ref_name::AbstractString)
     return oid_ptr[]
 end
 
+"""
+    GitHash(obj::GitObject)
+
+Gets the identifier (`GitHash`) of `obj`.
```
Comment (ararslan) [inline base/libgit2/oid.jl]: Get

File: base/libgit2/oid.jl [hunk 5]
```diff
@@ -61,10 +100,31 @@ function GitHash(repo::GitRepo, ref_name::AbstractString)
...
+"""
 function GitHash(obj::GitObject)
     GitHash(ccall((:git_object_id, :libgit2), Ptr{UInt8}, (Ptr{Void},), obj.ptr))
 end
 
+"""
+    GitShortHash(obj::GitObject)
+
+Gets a shortened identifier (`GitShortHash`) of `obj`. The minimum length (in characters)
```
Comment (ararslan) [inline base/libgit2/oid.jl]: Get

File: base/libgit2/oid.jl [hunk 6]
```diff
@@ -61,10 +100,31 @@ function GitHash(repo::GitRepo, ref_name::AbstractString)
...
+is determined by the `core.abbrev` config option, and will be of sufficient length to
```
Comment (ararslan) [inline base/libgit2/oid.jl]: Where can one access the value of said option?

File: base/libgit2/oid.jl (hunk 1 above)
Comment (simonbyrne) [inline base/libgit2/oid.jl]: In the spirit of Docathlon, feel free to edit directly.

Comment (ararslan) [inline base/libgit2/oid.jl]: Good call. 🙂

**Comment by simonbyrne** [comment]
Thanks!

**Comment by ararslan** [comment]
My pleasure. I don't know where to point people to access the `core.abbrev` thing you mentioned though.

**Pull request merged**